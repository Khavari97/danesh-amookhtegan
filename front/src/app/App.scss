@import 'assets/fonts/fonts';

* {
  margin: 0;
  font-family: "Shabnam", serif;
}

a {
  text-decoration: none;
}

//Slider css
.alice-carousel__dots {
  .alice-carousel__dots-item:not(.__custom).__active {
    background-color: #046B00;
  }
}


