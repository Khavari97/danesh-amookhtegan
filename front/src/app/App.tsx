import React from "react";
import "./App.scss";
import AppRouter from "./navigation/AppRouter";
import {SnackbarProvider} from 'notistack';
import {SnackbarUtils} from "./components/material-ui/Snackbar/SnackbarUtils";
import ThemeProvider from "./assets/styles/material/material";

export interface AppProps {
}

export interface AppState {
}

class App extends React.Component<AppProps, AppState> {
    constructor(props: AppProps) {
        super(props);
        this.state = {};
    }

    componentDidMount = () => {
    }

    render() {
        return (
            <SnackbarProvider maxSnack={3} anchorOrigin={{horizontal: 'right', vertical: 'bottom'}}>
                <SnackbarUtils/>
                <ThemeProvider>
                    <AppRouter/>
                </ThemeProvider>
            </SnackbarProvider>
        );
    }
}

export default App;
