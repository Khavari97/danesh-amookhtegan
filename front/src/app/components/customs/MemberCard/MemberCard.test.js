import React from "react"
import Enzyme, { shallow, configure, mount } from "enzyme";
import App from "./MemberCard";
import Adapter from '@wojtekmaj/enzyme-adapter-react-17'
import { Avatar, Grid, Link, Typography } from '@material-ui/core';
import Card from '@material-ui/core/Card';
import CardActions from '@material-ui/core/CardActions';
import CardContent from '@material-ui/core/CardContent';
import renderer from "react-test-renderer";

configure({ adapter: new Adapter() })

const setup = (props = {}) => {
  return shallow(<App {...props} />);
};

const findByTestAttr = (wrapper, val) => {
  return wrapper.find(`[data-test="${val}"]`);
};


describe('membercard component', () => {

  let wrapper = mount(<App />)

  it('render component check exists', () => {
    expect(wrapper.find(Avatar).exists()).toBe(true);
    expect(wrapper.find(Link).exists()).toBe(true);
    expect(wrapper.find(Card).exists()).toBe(true);
    expect(wrapper.find(CardActions).exists()).toBe(true);
    expect(wrapper.find(CardContent).exists()).toBe(true);
    expect(wrapper.find(Avatar).exists()).toBe(true);
  });
  it('render component check count of each of component', () => {
    expect(wrapper.find(Avatar).length).toBe(1);
    expect(wrapper.find(Link).length).toBe(1);
    expect(wrapper.find(Card).length).toBe(1);
    expect(wrapper.find(CardActions).length).toBe(1);
    expect(wrapper.find(CardContent).length).toBe(1);
  });

})
describe('input component', () => {
  let wrapper = mount(<App />)

  it('check set props', () => {

    wrapper.setProps({
      avatar: "avatar",
      firstName: 'atena',
      lastName: 'ganji',
      city: 'tehran',
      province: 'tehran',
      description: 'description',
      url: 'string'
    })

    expect(wrapper.props().avatar).toBe("avatar");
    expect(wrapper.props().firstName).toEqual("atena");
    expect(wrapper.props().lastName).toEqual("ganji");
    expect(wrapper.props().province).toEqual("tehran");
    expect(wrapper.props().city).toEqual("tehran");
    expect(wrapper.props().description).toEqual("description");
    expect(wrapper.props().url).toEqual("string");
  });
  it('snapshot', () => {
    const shot = <App />
    const snapshot = renderer.create(shot).toJSON();
    expect(snapshot).toMatchSnapshot();
  })
})

describe('input components', () => {
  let wrapper = setup()

  it('check render props in component', () => {

    wrapper.setProps({
      avatar: "avatar",
      firstName: 'atena',
      lastName: 'ganji',
      city: 'tehran',
      province: 'tehran',
      description: 'description',
      url: 'string'
    })

    expect(findByTestAttr(wrapper, 'name').text()).toBe("atena ganji");
    expect(findByTestAttr(wrapper, 'desc').text()).toBe("description")
    expect(findByTestAttr(wrapper, 'city').text()).toBe("tehran , tehran")
    expect(wrapper.find(Avatar).prop('src')).toBe("avatar");
    expect(wrapper.find(Link).prop('href')).toBe("string");
  });
})


