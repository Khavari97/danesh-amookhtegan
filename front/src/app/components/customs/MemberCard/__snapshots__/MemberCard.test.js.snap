// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`input component snapshot 1`] = `
<div>
  <div
    className="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      className="MuiCardContent-root makeStyles-root-1"
    >
      <div
        className="MuiGrid-root MuiGrid-container"
      >
        <div
          className="MuiGrid-root makeStyles-flexGrid-2 MuiGrid-item MuiGrid-grid-xs-12"
        >
          <div
            className="MuiAvatar-root MuiAvatar-circle MuiAvatar-colorDefault"
            style={
              Object {
                "height": 100,
                "width": 100,
              }
            }
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiAvatar-fallback"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
              />
            </svg>
          </div>
        </div>
        <div
          className="MuiGrid-root makeStyles-flexGrid-2 MuiGrid-item MuiGrid-grid-xs-12"
          data-test="name"
        >
          <p
            className="MuiTypography-root MuiTypography-body1"
          >
            undefined undefined
          </p>
        </div>
        <div
          className="MuiGrid-root makeStyles-flexGrid-2 MuiGrid-item MuiGrid-grid-xs-12"
        >
          <p
            className="MuiTypography-root MuiTypography-body1"
            data-test="city"
          >
            شهر و استان ثبت نشده
          </p>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
          style={
            Object {
              "direction": "rtl",
              "textAlign": "right",
            }
          }
        >
          <p
            className="MuiTypography-root MuiTypography-body1"
            data-test="desc"
            style={
              Object {
                "overflow": "hidden",
                "textOverflow": "ellipsis",
                "whiteSpace": "nowrap",
              }
            }
          >
            توضیحاتی موجود نیست
          </p>
        </div>
      </div>
    </div>
    <div
      className="MuiCardActions-root MuiCardActions-spacing"
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        }
      }
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-containedSizeSmall MuiButton-sizeSmall"
        disabled={false}
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          <a
            className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
            onBlur={[Function]}
            onFocus={[Function]}
            style={
              Object {
                "color": "white",
                "textDecoration": "none",
              }
            }
          >
            مشاهده رزومه
          </a>
        </span>
      </button>
    </div>
  </div>
</div>
`;
